pkg/manual_tests/user_default_database_and_role/test.md

```diff
@@ -0,0 +1,137 @@
+- Only role is granted to the user and the casing is matching.
+
+## Setup
+
+As a testing environment, I chose VSCode with the Snowflake extension for being able to quickly go between ACCOUNTADMIN account
+and the tested one. For testing account, I added this configuration for the tested user:
+
+```toml
+[connections.Test]          
+accountname = '***'
+username = '***'
+password= '***'
+host = '***'
+```
+
+The only thing that won't change between the tests is role and database, so we can create it beforehand:
+
+```sql
+CREATE ROLE TEST_ROLE;
+CREATE DATABASE TEST_DATABASE;
+```
+
+> Note: We have to use TYPE = LEGACY_SERVICE to be able to use 
+
+### 1. Either database or role is not present.
+
+1. Create user with the non-existing default database and role.
+
+```sql
+CREATE OR REPLACE USER TEST_USER
+    TYPE = LEGACY_SERVICE
+    LOGIN_NAME = 'login_name'
+    PASSWORD = 'password'
+    DEFAULT_NAMESPACE = 'NON_EXISTING_TEST_DATABASE'
+    DEFAULT_ROLE = 'NON_EXISTING_TEST_ROLE';
+```
+
+2. Log into the user (logs in successfully, but no database and role is selected in the context).
+
+### 2. Database and role are present, but not granted to the user.
+
+1. Replace the user with existing default database and role.
+
+```sql
+CREATE OR REPLACE USER TEST_USER
+    TYPE = LEGACY_SERVICE
+    LOGIN_NAME = 'login_name'
+    PASSWORD = 'password'
+    DEFAULT_NAMESPACE = 'TEST_DATABASE'
+    DEFAULT_ROLE = 'TEST_ROLE';
+```
+
+2. Log into the user (logs in successfully, but no database and role is selected in the context).
+   
+### 3. Database and role are granted to the user, but the casing is not matched.
+
+1. Replace the user with existing default database and role, but with lowercase names.
+
+```sql
+CREATE OR REPLACE USER TEST_USER
+    TYPE = LEGACY_SERVICE
+    LOGIN_NAME = 'login_name'
+    PASSWORD = 'password'
+    DEFAULT_NAMESPACE = 'test_database'
+    DEFAULT_ROLE = 'test_role';
+```
+
+2. Grant the role to the user and grant usage on the database to the role.
+
+```sql
+GRANT ROLE TEST_ROLE TO USER TEST_USER;
+GRANT USAGE ON DATABASE TEST_DATABASE TO ROLE TEST_ROLE;
+```
+
+3. Log into the user (logs in successfully, but no database and role is selected in the context).
+
+### 4. Database and role are granted to the user and the casing is matching.
+
+1. Replace the user with existing default database and role and exact casing.
+
+```sql
+CREATE OR REPLACE USER TEST_USER
+    TYPE = LEGACY_SERVICE
+    LOGIN_NAME = 'login_name'
+    PASSWORD = 'password'
+    DEFAULT_NAMESPACE = 'TEST_DATABASE'
+    DEFAULT_ROLE = 'TEST_ROLE';
+```
+
+2. Grant the role to the user and grant usage on the database to the role.
+
+```sql
+GRANT ROLE TEST_ROLE TO USER TEST_USER;
+GRANT USAGE ON DATABASE TEST_DATABASE TO ROLE TEST_ROLE;
+```
+
+3. Log into the user (logs in successfully and the database and role are selected in the context).
+
+### 5. Only role is granted to the user and the casing is matching.
```

Comment:
lets do three more where:
1. role matches and is granted
2. database access is given to the role
3. database in create user is lowercased (versus the uppercased created one)

second and third:
the same as the previous one but let's create lowercased db and:
- use lowercase in create user (second test)
- use lowercase in create user (third test)

Reply:
I guess the first one proposed is already test nr 3. I'll add other ones with lowercase role and database.

Reply:
Also added summary that summarizes the behavior of those two parameters, so it's clearer what the outcomes are.